Dungeon Party Join Info

TakoTheDev opened the pull request:
Doesnt get stats from the party you joined, since that kinda broke. but if people join your party it works

TakoTheDev committed:
Update CataCommand.kt

TakoTheDev committed:
deleted non working stuff

TakoTheDev committed:
Update Config.kt

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/commands/stats/impl/CataCommand.kt:
```diff
@@ -222,17 +222,13 @@ object CataCommand : StatCommand("skytilscata") {
                 append(':')
             }
             append("%02d".format(seconds))
-            if (nanoseconds != 0) {
```
What happened here

TakoTheDev replied:
Dont accept that file change since a function will be added to ChatListeners.kt

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/core/Config.kt:
```diff
@@ -110,6 +110,13 @@ object Config : Vigilant(File("./config/skytils/config.toml"), "Skytils", sortin
     var overrideReparty = true
+    
+    @Property(
+        type = PropertyType.SWITCH, name = "[WIP]Party Finder Stats",
```
Should follow the same style

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/core/Config.kt:
```diff
@@ -110,6 +110,13 @@ object Config : Vigilant(File("./config/skytils/config.toml"), "Skytils", sortin
+    @Property(
+        type = PropertyType.SWITCH, name = "[WIP]Party Finder Stats",
+        description = "Displays Stats about a Player who joined.",
+        category = "General", subcategory = "Party Finder"
```
Category should be dungeons?

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/core/Config.kt:
```diff
@@ -110,6 +110,13 @@ object Config : Vigilant(File("./config/skytils/config.toml"), "Skytils", sortin
+        description = "Displays Stats about a Player who joined.",
+        category = "General", subcategory = "Party Finder"
+    )
+    var partyFinderStats = true
```
Should not be default on

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -46,6 +58,35 @@ class ChatListener {
             return
         }
+        if(Skytils.config.partyFinderStats) {
+            val joinedd = dungeon_finder_join_pattern.matcher(unformatted)
```
Snake case is not convention but ok

TakoTheDev committed:
Discat_

Fixed a color code

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+            val armor = profileData.armor
+            val armorArray:ArrayList<List<String>> = ArrayList()
+            armor?.forEveryItem {
+                val armorLore = it.asMinecraft.getTooltip(mc.thePlayer,false)
```
Missing space

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                armorArray.add(armorLore)
+            }
+            armorArray.reverse()
+            for(armorPiece in armorArray) {
```
Missing space

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+            }
+            if (activePet != null) {
+                var rarity = "§f"
+                if (activePet.tier.toString().uppercase() == "UNCOMMON") {
```
What is this? Consider ItemRarity

TakoTheDev replied:
Will be fixed

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                    rarity = "§d"
+                }
+
+                component.append(UTextComponent(rarity + activePet.type.toString().lowercase().replace("_", " ")
```
Type should already be a string?

TakoTheDev replied:
Ide complaint but will try without

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                    .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()}
+                        + "\n\n")
+                    .setHoverText( if(activePet.heldItem == null) {"§cNo Pet Item"} else "§b" +
+                            activePet.heldItem.toString().lowercase().replace("pet_item_", "").replace("_", " ")
```
Held item should already be a string

TakoTheDev replied:
Same as above

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                        + "\n\n")
+                    .setHoverText( if(activePet.heldItem == null) {"§cNo Pet Item"} else "§b" +
+                            activePet.heldItem.toString().lowercase().replace("pet_item_", "").replace("_", " ")
+                                .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()}
```
Why is there a locale supplied here

TakoTheDev replied:
Ide was crying about me just doing it normally and suggested me to do this

My-Name-Is-Jeff commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+            //Stonk etc. stuff
+            val inventory = profileData.inventory
+            if (inventory != null) {
+                var hasStonk = false
```
These are all unnecessary

Sychic replied:
Agree

TakoTheDev replied:
agree and Will be fixed

TakoTheDev replied:
fixed this part a lot

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                        hasShadowFury = true
+                    } else if (itemName.contains("Flower of Truth")) {
+                        hasFlowerOfTruth = true
+                    }
```
This should be a when block

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                        append("§cFot: §a§l●\n")
+                    } else {
+                        append("§2Giant's Sword: §c§l●\n")
+                    }
```
This should all be a when block, also i'd consider gs a mage weapon and not bers

Tricked-dev commented:
why did you make 3 pr’s lol

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                var hasShadowFury = false
+                var hasFlowerOfTruth = false
+
+                inventory.forEveryItem {
```
It might be better to just convert the inventory to a set and then intersect with a set of items you want to look for

TakoTheDev replied:
thank you :D

TakoTheDev committed:
changed default state to false(off)

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                .append("§aTotal Secrets Found: §l§6${NumberUtil.nf.format(secrets)}\n\n")
+                .append(UTextComponent("§cKick $username\n").setHoverText("§cClick to Paste the Command to Kick $username.").setClick(ClickEvent.Action.SUGGEST_COMMAND, "/p kick $username"))
+                .append("&2&l-----------------------------")
+                .chat()
```
Maybe it'd be worth making it mutable to later delete? since it does kinda spam your chat with a giant wall of text

Sychic replied:
Additionally it could delete the text when you kick them

TakoTheDev replied:
Should i maybe auto delete it after 1 or 2 Minutes?

TakoTheDev replied:
Made mutable but not sure how to delete it afterwards

Sychic replied:
Make it run a command which calls delete with the corresponding chat id (i think it's public though i forget)

TakoTheDev committed:
Fixed Category to be Dungeons not General

TakoTheDev committed:
Fixed 2 Spaces missing

TakoTheDev committed:
added time format function

and changed it to not use catacommands one. Catacommand changes can now be discarded

TakoTheDev committed:
made function private again

TakoTheDev committed:
removed unnecessary spaces

TakoTheDev committed:
reverted cataCommand changes

TakoTheDev committed:
Add files via upload

TakoTheDev committed:
Copy pasted my code

TakoTheDev committed:
Update ChatListener.kt

TakoTheDev committed:
fixed another locale

TakoTheDev committed:
fixed chat color for kick button

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -279,23 +281,21 @@ class ChatListener {
-                val armorLore = it.asMinecraft.getTooltip(mc.thePlayer,false)
+                val armorLore = it.asMinecraft.getTooltip(mc.thePlayer, false)
                 armorArray.add(armorLore)
             }
```
This should just be a map from the items, you shouldn't be doing it this way

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -279,23 +281,21 @@ class ChatListener {
             }
             armorArray.reverse()
-            for(armorPiece in armorArray) {
+            for (armorPiece in armorArray) {
```
Use `joinToString` instead of a for loop

TakoTheDev replied:
thanks, didn't know that existed in kotlin

TakoTheDev replied:
Half fixed

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -312,24 +312,20 @@ class ChatListener {
+                    ItemRarity.EPIC.rarityName -> rarity = "§5"
+                    ItemRarity.LEGENDARY.rarityName -> rarity = "§6"
+                    ItemRarity.MYTHIC.rarityName -> rarity = "§d"
                 }
```
Item rarities should have corresponding chat colors, instead maybe use `Enum.valueOf()`

TakoTheDev replied:
You can use .baseColor, has already been fixed just not pr'ed

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -312,24 +312,20 @@ class ChatListener {
-                            activePet.heldItem.toString().lowercase().replace("pet_item_", "").replace("_", " ")
-                                .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()}
+                            activePet.heldItem!!.lowercase().replace("pet_item_", "")
+                                .replace("_", " ").replaceFirstChar { it.uppercase() }
```
You can use kotlin's nullable handling to add a default instead of using an if/else

TakoTheDev committed:
Made UMessage component mutable

When kick pressed, delete the message will be added later on

TakoTheDev committed:
Fixed PetItem Stuff

TakoTheDev committed:
Fix Armor to use JoinToString

TakoTheDev committed:
Some in progress stuff

TakoTheDev committed:
Fixed some stuff

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -21,12 +21,14 @@ import gg.essential.universal.UChat
 import net.minecraftforge.client.event.ClientChatReceivedEvent
 import net.minecraftforge.fml.common.eventhandler.EventPriority
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import scala.Console
```
Remove this line

TakoTheDev replied:
Fixed

TakoTheDev committed:
Added Spirit Pet Support

Sychic commented on src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt:
```diff
@@ -297,132 +299,114 @@ class ChatListener {
-                        hasFlowerOfTruth = true
-                    }
+                    inv.add(it.asMinecraft.displayName)
                 }
```
Use `Inventory::items` and the convert it to a set to intersect

TakoTheDev committed:
Just uploaded latest

Dont think anything new just to make sure i dont accidently delete it or so

Sychic commented:
i can't read and thought it said clone

My-Name-Is-Jeff committed:
tako pro coder

Sychic committed:
Capitalize pet item nicely

Sychic committed:
Look for high efficiency picks too

Sychic committed:
Use right casing for pet

Sychic committed:
Always show spirit pet

Sychic committed:
Change username formatting

Sychic committed:
Kotlin-ify

Sychic commented on src/main/kotlin/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt:
```diff
@@ -0,0 +1,274 @@
+            val username = match.groups["name"]?.value ?: return
+            if (username == mc.thePlayer.name) return
+            Skytils.launch {
+                val uuid = try {
+                    Skytils.hylinAPI.getUUIDSync(username)
+                } catch (e: AshconException) {
+                    UChat.chat("§cFailed to get UUID, reason: ${e.message}")
+                    return@launch
+                }
+                val profile = try {
+                    Skytils.hylinAPI.getLatestSkyblockProfileForMemberSync(uuid)
+                } catch (e: HypixelAPIException) {
+                    UChat.chat("§cUnable to retrieve profile information: ${e.message}")
+                    return@launch
+                }
```
Probably want to replace these with coroutine exception handling

Sychic commented on src/main/kotlin/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt:
```diff
@@ -0,0 +1,274 @@
+            return
+        }
+
+        try {
```
Still don't know why this try catch is here so I left it

Sychic replied:
leaving this here as a safety net

Sychic committed:
Fix missing master cata data

Sychic committed:
Use Hylin request catching

My-Name-Is-Jeff committed:
Add pet level
